plugin/kprom/kprom.go

```diff
@@ -1,260 +1,299 @@
+		Namespace:   namespace,
+		Subsystem:   subsystem,
+		ConstLabels: constLabels,
+		Name:        "read_bytes_total",
+		Help:        "Total number of bytes read, by nodeid",
+	}, []string{"nodeid"})
+
+	m.readWaits = factory.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace:   namespace,
+		Subsystem:   subsystem,
+		ConstLabels: constLabels,
+		Name:        "read_wait_latency_seconds",
+		Help:        "Time spent waiting to read from Kafka, by nodeid",
+		Buckets:     buckets,
+	}, []string{"nodeid"})
+
+	m.readTimings = factory.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace:   namespace,
+		Subsystem:   subsystem,
+		ConstLabels: constLabels,
+		Name:        "read_latency_seconds",
+		Help:        "Time spent reading from Kafka, by nodeid",
+		Buckets:     buckets,
+	}, []string{"nodeid"})
+
+	// produces & consumes
+
+	m.produceBatchesUncompressed = factory.NewCounterVec(prometheus.CounterOpts{
+		Namespace:   namespace,
+		Subsystem:   subsystem,
+		ConstLabels: constLabels,
+		Name:        "produce_bytes_uncompressed_total",
+		Help:        "Total number of uncompressed bytes produced, by nodeid and topic",
+	}, []string{"nodeid", "topic"})
+
+	m.produceBatchesCompressed = factory.NewCounterVec(prometheus.CounterOpts{
+		Namespace:   namespace,
+		Subsystem:   subsystem,
+		ConstLabels: constLabels,
+		Name:        "produce_bytes_compressed_total",
+		Help:        "Total number of compressed bytes produced, by nodeid and topic",
+	}, []string{"nodeid", "topic"})
+
+	m.produceBatchesRecords = factory.NewCounterVec(prometheus.CounterOpts{
+		Namespace:   namespace,
+		Subsystem:   subsystem,
+		ConstLabels: constLabels,
+		Name:        "produce_records_total",
+		Help:        "Total number of records produced, by nodeid",
+	}, []string{"nodeid", "topic"})
+
+	m.fetchBatchesUncompressed = factory.NewCounterVec(prometheus.CounterOpts{
+		Namespace:   namespace,
+		Subsystem:   subsystem,
+		ConstLabels: constLabels,
+		Name:        "fetch_bytes_uncompressed_total",
+		Help:        "Total number of uncompressed bytes fetched, by nodeid, by topic",
+	}, []string{"nodeid", "topic"})
+
+	m.fetchBatchesCompressed = factory.NewCounterVec(prometheus.CounterOpts{
+		Namespace:   namespace,
+		Subsystem:   subsystem,
+		ConstLabels: constLabels,
+		Name:        "fetch_bytes_compressed_total",
+		Help:        "Total number of compressed bytes actually fetched, by nodeid, by topic",
+	}, []string{"nodeid", "topic"})
+
+	m.fetchBatchesRecords = factory.NewCounterVec(prometheus.CounterOpts{
+		Namespace:   namespace,
+		Subsystem:   subsystem,
+		ConstLabels: constLabels,
+		Name:        "fetch_records_total",
+		Help:        "Total number of records fetched, by nodeid",
+	}, []string{"nodeid", "topic"})
+
+	// e2e timing
+
+	m.e2eTimings = factory.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace:   namespace,
+		Subsystem:   subsystem,
+		ConstLabels: constLabels,
+		Name:        "e2e_broker_latency_seconds",
```

Comment:
I'd rename that to node_latency rather than broker_latency so that it's consistent with the rest of the client naming. Also we use the label "nodeid".

Reply:
💯